vignettes/advs.Rmd

```diff
@@ -0,0 +1,404 @@
+
+```{r, warning=FALSE, message=FALSE}
+library(admiral)
+library(admiralmetabolic)
+library(pharmaversesdtm)
+library(dplyr, warn.conflicts = FALSE)
+```
+
+# Programming Workflow
+
+-   [Read in Data](#readdata)
+-   [Assign `PARAMCD`, `PARAM`, `PARAMN`, `PARCAT1`](#paramcd)
+-   [Derive Additional Parameters (e.g. `BMI` for `ADVS`)](#derive_param)
+-   [Common Metabolic Endpoints](#common_endpoints)
+    -   [Derive Categorization Variables (`AVALCATx`, `BASECATx`)](#cat)
+    -   [Derive Criterion Variables (`CRITy`, `CRITyFL`, `CRITyFLN`)](#crit_vars)
+-   [Remaining ADVS Set-up](#advs_end)
+
+## Read in Data {#readdata}
+
+To start, all data frames needed for the creation of `ADVS` should be
+loaded into the global environment. This will be a company specific process. Some
+of the data frames needed may be `VS` and `ADSL`.
+
+```{r, message=FALSE, warning=FALSE}
+dm_metabolic <- admiralmetabolic::dm_metabolic
+vs_metabolic <- admiralmetabolic::vs_metabolic
+admiral_adsl <- admiral::admiral_adsl
+
+dm <- convert_blanks_to_na(dm_metabolic)
+vs <- convert_blanks_to_na(vs_metabolic)
+admiral_adsl <- convert_blanks_to_na(admiral_adsl)
+```
+
+Within this vignette, `DM` is used as the basis for `ADSL`:
+
+```{r eval=TRUE}
+# Retrieve required variables from admiral ADSL for this vignette that are not present in DM dataset
+adsl <- dm %>%
+  select(-DOMAIN) %>%
+  mutate(TRT01P = ARM, TRT01A = ACTARM) %>%
+  left_join(admiral_adsl %>% select(USUBJID, TRTSDT, TRTEDT), by = "USUBJID")
+```
+
+The following steps are to merge `ADSL` variables with the source data and
+derive the usual `ADVS` analysis variables. Note that only the sections required for this vignette are covered in the following steps. To get a detailed guidance on all the steps, refer the `{admiral}` [Creating a BDS Finding ADaM vignette](https://pharmaverse.github.io/admiral/articles/bds_finding.html).
+
+Join `ADSL` to your `VS` domain. Only the `ADSL` variables used for derivations are selected at this step.
+```{r, eval=TRUE}
+adsl_vars <- exprs(TRTSDT, TRTEDT, TRT01P, TRT01A)
+
+advs <- derive_vars_merged(
+  vs,
+  dataset_add = adsl,
+  new_vars = adsl_vars,
+  by_vars = exprs(STUDYID, USUBJID)
+)
+```
+
+The function `derive_vars_dt()` can be used to derive `ADT`.
+
+```{r, eval=TRUE}
+advs <- derive_vars_dt(advs, new_vars_prefix = "A", dtc = VSDTC)
+```
+
+Once `ADT` is derived, the function `derive_vars_dy()` can be used to derive `ADY`
+
+```{r, eval=TRUE}
+advs <-
+  derive_vars_dy(advs, reference_date = TRTSDT, source_vars = exprs(ADT))
+```
+
+## Assign `PARAMCD`, `PARAM`, `PARAMN`, `PARCAT1` {#paramcd}
+
+The next step is to assign parameter level values such as `PARAMCD`, `PARAM`,
+`PARAMN`, `PARCAT1`, etc. For this, a lookup can be created based on
+the SDTM `--TESTCD` value to join to the source data.
+The only difference here would be the the additional parameters
+collected in metabolic trials - `BMI`, `HIPCIR` and `WSTCIR`.
```

Comment:
```suggestion
One key distinction to note is the additional parameters collected in metabolic trials - `BMI`, `HIPCIR`, and `WSTCIR`.
```
The idea was to highlight the additional parameters collected in this vignette that are not there in standard one and display them in this section. To not cause confusion , can we rephrase it as suggested above ?
@AndersAskeland